test/topology/test_gossiper_orphan_remover.py

```diff
@@ -0,0 +1,42 @@
+import time
+import pytest
+import logging
+from test.pylib.manager_client import ManagerClient
+from test.topology.conftest import skip_mode
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_crashed_node_substitution(manager: ManagerClient):
+    servers = await manager.running_servers()
+
+    try:
+        await manager.server_add(config={
+            'error_injections_at_startup': ['crash_before_group0_join']
+        })
+    except Exception as e:
+        logger.info(f"exception raised due to : {e}")
+    else:
+        raise Exception("error injected server startup should result in a failure, but unexpectedly succeeded")
+
+    live_eps = await manager.api.client.get_json("/gossiper/endpoint/live", host=servers[0].ip_addr)
+    down_eps = await manager.api.client.get_json("/gossiper/endpoint/down", host=servers[0].ip_addr)
+
+    gossiper_eps = live_eps + down_eps
+    post_crash_servers = await manager.running_servers()
+    server_eps = [s.ip_addr for s in post_crash_servers]
+
+    assert len(gossiper_eps) == (len(server_eps)+1)
+
+    orphan_ip = [ip for ip in gossiper_eps if ip not in server_eps][0]
+
+    # Allows backgound fiber to finish removing orphan nodes.
+    time.sleep(75)
```

Comment:
Please figure out how to do it in a way that is:
- faster (should be almost instantaneous in most test runs)
- and at the same time reliable (i.e. the test knows for 100% that the removal happened and the checks below should succeed) -- so it doesn't rely on timing

Reply:
Generally, sleeping for arbitrary amount of time results in tests that are both flaky and slow.

Reply:
Done, now everything is instantaneous